### PR TITLE
more clippy fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_typeck/src/check/expr.rs [hunk 1]
```diff
@@ -2058,8 +2058,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     ) -> Option<(&Vec<ty::FieldDef>, SubstsRef<'tcx>)> {
         debug!("get_field_candidates(span: {:?}, base_t: {:?}", span, base_t);
 
-        let mut autoderef = self.autoderef(span, base_t);
-        while let Some((base_t, _)) = autoderef.next() {
+        let autoderef = self.autoderef(span, base_t);
+        for (base_t, _) in autoderef {
```
Comment (jyn514) [inline compiler/rustc_typeck/src/check/expr.rs:2062]: You don't need a temporary here I think.
```suggestion
        for (base_t, _) in self.autoderef(span, base_t) {
```

**Comment by jyn514** [comment]
@bors try @rust-timer queue

r=me assuming perf isn't too bad :)

**Comment by matthiaskrgr** [comment]
removed the `autoderef` var.

@bors r=jyn514

**Comment by jyn514** [comment]
> GitHub Actions has encountered an internal error when running your job.

🤦 

@bors retry

**Pull request merged**